tcms/tests/__init__.py

```diff
@@ -227,3 +230,47 @@ def setUpTestData(cls):
         cls.execution_4 = executions[3]
         cls.execution_5 = executions[4]
         cls.execution_6 = executions[5]
+
+
+class GetListOffAllPossiblePermissions(BaseCommand):
+
+    # Helper class for getting all possible permissions
+    # Taken from:
+    # timonweb.com/posts/how-to-get-a-list-of-all-user-permissions-available-in-django-based-project/
+
+    def handle(self, *args, **options):
+        permissions = set()
+
+        tmp_superuser = get_user_model()(
+            is_active=True,
+            is_superuser=True
+        )
+
+        for backend in auth.get_backends():
+            if hasattr(backend, "get_all_permissions"):
+                permissions.update(backend.get_all_permissions(tmp_superuser))
+
+        sorted_list_of_permissions = sorted(list(permissions))
+
+        return sorted_list_of_permissions
+
+
+class CreateTestUsers(LoggedInTestCase):
+    """Base test class for user creation"""
+
+    @classmethod
+    def setUpTestData(cls):
+        super(CreateTestUsers, cls).setUpTestData()
+
+    def createUserWithNoPermissions(self):
+        parameters = GetListOffAllPossiblePermissions.handle()
+        for perm in parameters():
+            remove_perm_from_user(self.tester, perm)
+
+        return self.tester
```

Comment:
When `self.tester` is initially created it doesn't have any permissions assigned. When doing the actual testing, e.g. inside of the test method one will write something like:

```
def test_with_permission(self):
    self.has_only('permission.label')

    ... actual testing code ....

def test_without_permission(self):
    self.is_missing_only('permission.label')

    ... actual testing code ....
```

These methods can actually be called via super() in the inherited classes to make sure the testing environment is properly set up and the actual implementation deals with calling backend views/API and asserting on the expected results.